src/pudl/transform/ferc1.py

```diff
@@ -2759,6 +2820,118 @@ def normalize_metadata_xbrl(
+        other debt records, one of which contains one null record.
+
+        This method is part of the DBF processing because we want to be able to
+        hard-code a specific value of ``utility_id_ferc1_dbf`` and those IDs are no
+        longer available later in the process. I think.
+        """
+        if 1999 in df.report_year.unique():
+            pks = ["utility_id_ferc1", "report_year", "asset_type"]
+            bespoke_dupe_mask = (
+                df.duplicated(subset=pks, keep=False)
+                & (df.report_year == 1999)
+                & (df.utility_id_ferc1_dbf == 165)
+            )
+            bespoke_dupe_null_mask = bespoke_dupe_mask & (df.starting_balance.isnull())
+            # checks to make sure the drop is targeted as expected
+            if len(df[bespoke_dupe_mask]) != 2:
+                raise AssertionError(
+                    "Expected to find two duplicate records from 1999 with "
+                    f"utility_id_ferc1_dbf 165, but found {len(df[bespoke_dupe_mask])}"
+                )
+            if len(df[bespoke_dupe_null_mask]) != 1:
+                raise AssertionError(
+                    "Expected to find one record from 1999 with utility_id_ferc1_dbf "
+                    "165 and null starting_balance, but found "
+                    f"{len(df[bespoke_dupe_null_mask])}"
+                )
+            logger.debug(f"{self.table_id.value}: Dropping one duplicate from 1999.")
+            df = df[~bespoke_dupe_null_mask].copy()
+        return df
+
+    def process_dbf(self, raw_dbf: pd.DataFrame) -> pd.DataFrame:
+        """Drop targeted duplicates in the DBF data so we can use FERC respondent ID.
+
+        First employ the standard :meth:`process_dbf`. Then deal with the duplicate
+        records. This is done by first running :meth:`drop_invalid_rows` pre-emtively to
+        remove any possible rows with duplicate primary key values that have nulls in
+        the data fields. Then we drop a targeted duplicate. Then we run
+        :meth:`drop_duplicate_rows_dbf`, which drops any records with duplicate primary
+        key values as long as those records also contain duplicate data.
+        """
+        return (
+            super()
+            .process_dbf(raw_dbf)
+            .pipe(self.drop_invalid_rows)
+            .pipe(self.targeted_drop_duplicates_dbf)
+            .pipe(self.drop_duplicate_rows_dbf)
+        )
+
+    def normalize_metadata_xbrl(
+        self, xbrl_fact_names: list[str] | None
+    ) -> pd.DataFrame:
+        """Normalize the metadata from the XBRL taxonomy +."""
+        meta = (
+            super()
+            .normalize_metadata_xbrl(xbrl_fact_names)
+            .assign(
+                xbrl_factoid=lambda x: x.xbrl_fact_name,
+            )
+        )
+        # Save the normalized metadata so it can be used by other methods.
+        self.xbrl_metadata_normalized = meta
+        return meta
```

Comment:
100%! I think @zaneselvans is handling this as he is integrating the new metadata.

I believe the outcome of it will be removing the need for all of these direct copy/paste methods that are just adding a column with a different name.